Prose/ProseLib/Sources/AuthenticationFeature/MFA6DigitsView.swift

```diff
@@ -0,0 +1,275 @@
+        }
+        // Hide fake fields in accessibility
+        .accessibilityHidden(true)
+        // Add a real `TextField` backing the interactions and accessibility
+        .background {
+            TextField(l10n.Form.OneTimeCode.placeholder, text: viewStore.binding(\.$code))
+                .focused($focusedField, equals: .textField)
+                .textContentType(.oneTimeCode)
+                .opacity(0)
+        }
+        .font(.system(size: 24))
+        .disabled(viewStore.isLoading)
+        .opacity(viewStore.isLoading ? 0.5 : 1)
+    }
+
+    func mainButton(viewStore: ViewStore<State, Action>) -> some View {
+        Button { actions.send(.mainButtonTapped) } label: {
+            Text(l10n.ConfirmButton.title)
+                .frame(minWidth: 196)
+        }
+        .disabled(!viewStore.isMainButtonEnabled)
+    }
+
+    func digit(_ digit: String, selected: Bool) -> some View {
+        ZStack {
+            RoundedRectangle(cornerRadius: 6)
+                .fill(Color(nsColor: .controlBackgroundColor))
+                .shadow(radius: 0.75, y: 0.5)
+            if selected {
+                RoundedRectangle(cornerRadius: 6)
+                    .strokeBorder(Color.accentColor.opacity(0.75), lineWidth: 3)
+            }
+            if digit.isEmpty {
+                Text("0")
+                    .foregroundColor(.secondary)
+            } else {
+                Text(digit)
+            }
+        }
+        .frame(width: 48, height: 56)
+    }
+
+    func footer(viewStore: ViewStore<State, Action>) -> some View {
+        HStack(spacing: 16) {
+            Button(l10n.Footer.CannotGenerateCode.title) { actions.send(.showPopoverTapped(.cannotGenerateCode)) }
+                .popover(
+                    unwrapping: viewStore.binding(\.$popover),
+                    case: /State.Popover.cannotGenerateCode,
+                    content: { _ in Self.cannotGenerateCodePopover() }
+                )
+            Divider().frame(height: 18)
+            Button(l10n.Footer.NoAccount.title) { actions.send(.showPopoverTapped(.noAccount)) }
+                .popover(
+                    unwrapping: viewStore.binding(\.$popover),
+                    case: /State.Popover.noAccount,
+                    content: { _ in BasicAuthView.noAccountPopover() }
+                )
+        }
+        .buttonStyle(.link)
+    }
+
+    static func cannotGenerateCodePopover() -> some View {
+        GroupBox(l10n.Footer.CannotGenerateCode.title) {
+            Text(l10n.Footer.CannotGenerateCode.content)
+        }
+        .groupBoxStyle(PopoverGroupBoxStyle())
+    }
+}
+
+// MARK: - The Composabe Architecture
+
+// MARK: Reducer
+
+public let mfa6DigitsReducer: Reducer<
+    MFA6DigitsState,
+    MFA6DigitsAction,
+    AuthenticationEnvironment
+> = Reducer { state, action, environment in
+    switch action {
+    case .onAppear, .onGlobalTapGesture:
+        state.focusedField = .textField
+
+    case .mainButtonTapped:
```

Comment:
Nitpick: `submitButtonTapped` would be nicer than `mainButtonTapped` since it also carries what the button does not only its placement in the form. Makes it a little easier to comprehend.

Reply:
Yeah, ok. I also wanted to change the button in the login screen, as it is called `loginButton` and it does cancellation too…